File: FinalBootMVC/src/main/java/com/springboot/pkg/model/Ordinateur.java
package com.springboot.pkg.model;

import java.util.Date;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToOne;

@Entity
@DiscriminatorValue("O")
public class Ordinateur extends Materiel {
	
	private String processeur;
	private Integer ram;
	private Integer hdd;
	private Date anneeAchat;
	
	@OneToOne
	private Stagiaire stagiaire;
	
	public Ordinateur() {
		
	}
	
	public Ordinateur(String processeur, Integer ram, Integer hdd, Date anneeAchat) {
		this.processeur = processeur;
		this.ram = ram;
		this.hdd = hdd;
		this.anneeAchat = anneeAchat;
	}

	public String getProcesseur() {
		return processeur;
	}

	public void setProcesseur(String processeur) {
		this.processeur = processeur;
	}

	public Integer getRam() {
		return ram;
	}

	public void setRam(Integer ram) {
		this.ram = ram;
	}

	public Integer getHdd() {
		return hdd;
	}

	public void setHdd(Integer hdd) {
		this.hdd = hdd;
	}

	public Date getAnneeAchat() {
		return anneeAchat;
	}

	public void setAnneeAchat(Date anneeAchat) {
		this.anneeAchat = anneeAchat;
	}

	public Stagiaire getStagiaire() {
		return stagiaire;
	}

	public void setStagiaire(Stagiaire stagiaire) {
		this.stagiaire = stagiaire;
	}
	
	
	

}

File: FinalAngular/src/app/model/ordinateur.ts
import { Materiel } from './materiel';

export class Ordinateur extends Materiel {

    constructor(code: number, cout?: number, private _processeur?: string, private _hdd?: number,
         private _ram?: number, private _anneeAchat?: Date) {
        super(code, cout);
    }
    public get processeur(): string {
        return this._processeur;
    }
    public set processeur(value: string) {
        this._processeur = value;
    }
    public get hdd(): number {
        return this._hdd;
    }
    public set hdd(value: number) {
        this._hdd = value;
    }
    public get ram(): number {
        return this._ram;
    }
    public set ram(value: number) {
        this._ram = value;
    }
    public get anneeAchat(): Date {
        return this._anneeAchat;
    }
    public set anneeAchat(value: Date) {
        this._anneeAchat = value;
    }
}
